staking/effective/calculate.go

```diff
@@ -57,8 +75,14 @@ func median(stakes []SlotPurchase) numeric.Dec {
 	const isEven = 0
 	switch l := len(stakes); l % 2 {
 	case isEven:
-		return stakes[(l/2)-1].Dec.Add(stakes[(l/2)+1].Dec).Quo(two)
+		left := (l / 2) - 1
+		right := (l / 2)
+		msg := fmt.Sprintf("Len(stakes) is even. Left = %d. Right = %d.", left, right)
```

Comment:
this call to Sprintf is likely very expensive, relies on reflection. Better just make the chain of the debugger logger, that is, .Info().Uint("left", left).Uint("right", right)